File: experiment/order_of_if/positive_first.rb
def greeting(user)
  if user[:title]
    name = "#{user[:title]} #{user[:last_name]}"
  else
    name = user[:first_name]
  end

  'Dear ' + name
end

user = {
  title: 'Miss',
  first_name: 'Jane',
  last_name: 'Marple'
}

result = greeting user

File: experiment/order_of_if/positive_first.py
def greeting(user):
  if user.get('title'):
    name = user.get('title') + " " + user.get('last_name')
  else:
    name = user.get('first_name')

  return "Dear " + name

user = {
  'title': 'Miss',
  'first_name': 'Jane',
  'last_name': 'Marple'
}

result = greeting(user)